packages/eslint-plugin/src/rules/no-extra-semi.ts

```diff
@@ -16,7 +16,9 @@ export default util.createRule<Options, MessageIds>({
     },
     fixable: 'code',
     schema: baseRule.meta.schema,
-    messages: baseRule.meta.messages,
+    messages: baseRule.meta.messages ?? {
+      unexpected: 'Unnecessary semicolon.',
+    },
```

Comment:
this rule has been using message ids since 4.17, so we don't need to fallback here.

Reply:
reverted this change